src/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1180,6 +1182,48 @@ protected SAMBinaryTagAndValue getBinaryAttributes() {
     }
 
     /**
+     * Interval covering the aligned, unclipped, bases in this read as reported by getAlignmentStart() and getAlignmentEnd()
+     * @return the aligned interval or null if getReadUnmappedFlag()
+     */
+    public SimpleInterval getLocation() {
+        if (getReadUnmappedFlag()) {
+            return null;
+        } else {
+            return new SimpleInterval(mReferenceName, mAlignmentStart, mAlignmentEnd);
```

Comment:
This should probably use ```getAlignmentEnd``` instead of ```mAlignmentEnd```, since the later may be unset for mapped reads until the former is called.  It is probably better practice to use the getters in these calls anyhow.